File: Data Structures/Trees/Trie/trie.java
class TrieNode{
    TrieNode child[] = new TrieNode[26];
    boolean endWord;

    TrieNode(){
        for(int i=0; i<26; i++){
            child[i] = null;
        }
        endWord = false;
    }
};


public class trie {
    public static void insert_key(TrieNode root, String word){
        for(int i = 0; i < word.length(); i++){
            int index = (int)word.charAt(i) - (int)'a';
            if(root.child[index] == null){
                TrieNode temp = new TrieNode();
                root.child[index] = temp;
            }
            root = root.child[index];
        }
        root.endWord = true;
    };

    public static boolean search_key(TrieNode root, String word){
        for(int i = 0; i < word.length(); i++){
            int index = (int)word.charAt(i) - (int)'a';
            if(root.child[index] == null){
                return false;
            }
            root = root.child[index];
        }
        return root.endWord;
    };

    public static void main(String args[]){
        TrieNode root = new TrieNode();
        String arr[] = {"and", "ant", "do", "geek", "dad", "ball"};
        for(String word :arr)
            insert_key(root, word);
    
        String search_keys[] = {"do", "gee", "bat"};
        for (String s : search_keys){
            System.out.printf("Key : %s", s);
            if (search_key(root, s)){
                System.out.println(" Present") ;
            }
            else{
                System.out.println(" Not Present");
            }
        }
    }
}

File: Data Structures/Trees/Trie/trie.py
class TrieNode:
    def __init__(self) -> None:   
        self.child = [None] * 26
        self.endWord = False
    
def insert_key(root, word):
    for i in word:
        index = ord(i) - ord('a')
        if root.child[index] is None:
            temp = TrieNode()
            root.child[index] = temp
        root = root.child[index]
    root.endWord = True

def search_key(root, word):
    for i in word:
        index = ord(i) - ord('a')
        if root.child[index] is None:
            return False
        root = root.child[index]
    return root.endWord
    
if __name__ == "__main__":
    root = TrieNode()
    arr = ["and", "ant", "do", "geek", "dad", "ball"]
    for s in arr:
        insert_key(root, s)

    search_keys = ["do", "gee", "bat"]
    for s in search_keys:
        print(f"Key : {s}")
        if search_key(root, s):
            print("Present")
        else:
            print("Not Present")